projects/control-service/projects/model/apidefs/datajob-api/api.yaml

```diff
@@ -900,6 +900,57 @@ paths:
         in: path
         required: true
 
+  '/data-jobs/for-team/{team_name}/jobs/{job_name}/deployments/{deployment_id}/secrets':
+    summary: |
+      Data Job Secrets API.
+    get:
+      tags:
+        - Data Jobs Secrets
+      responses:
+        '200':
+          $ref: '#/components/responses/Secrets'
+      operationId: dataJobSecretsRead
+      summary: Get Data Job secrets.
+      description: <i>(Introduced in v1.0)</i>
+    put:
+      requestBody:
+        content:
+          application/json:
+            schema:
+              $ref: '#/components/schemas/DataJobSecrets'
+        required: true
+      tags:
+        - Data Jobs Secrets
+      responses:
+        '201':
+          description: Created successfully.
+        '204':
+          description: Updated successfully.
+        '413':
+          description: Payload too large. Combined size of secrets is larger than 1 MB.
```

Comment:
Is hardcoding resource requirements in the API a good idea? By design, is the secrets api extensible to use different backing storage and configurations?

Reply:
Fixed.